app/models/openscap_result.rb

```diff
@@ -0,0 +1,68 @@
+require 'openscap'
+require 'openscap/ds/arf'
+require 'openscap/xccdf/benchmark'
+
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  OpenSCAP.oscap_init
+
+  def html
+    return unless binary_blob && binary_blob.binary
+    with_oscap_file(binary_blob.binary) do |arf_path|
```

Comment:
We should be able to take ARF from memory. 

https://github.com/OpenSCAP/ruby-openscap/blob/master/lib/openscap/ds/arf.rb#L25
https://github.com/OpenSCAP/ruby-openscap/blob/master/lib/openscap/source.rb#L56

Something like this should just work:
```
OpenSCAP::DS::Arf.new(:content => raw_memory, :length => raw_memory_length, :path => 'my_arbitrary_arf_name.xml')
```

The `:path` you pass in, can be used by some error messages. However, a file at that location will not be touched. Some older version of OpenSCAP use the name to understand if you pass bzip stream (endswith(.xml.bz2)) or xml stream endswith('.xml'). The newer version of OpenSCAP do not care about filename, and just parse first few bytes to see, whether it is a bzip2.